### ISSUE TITLE
Error when height width not specified

### ISSUE THREAD

**Issue by 212361828** [PredixDev/px-vis#96]
px-vis 1.2.1

Dashboard team is seeing a console error with the px-vis-timeseries component when creating a new component that has timeseries as one of it's elements.  More specifically the following error is seen when removing then adding the timeseries component:

> Uncaught TypeError: Cannot read property 'push' of null
>     at HTMLElement.generateRandomID (px-vis-behavior-common.html:1885)
>     at HTMLElement._drawElement (px-vis-clip-path.html:94)
>     at HTMLElement._complexObserverEffect (polymer.html:1613)
>     at HTMLElement._effectEffects (polymer.html:1448)
>     at HTMLElement._propertySetter (polymer.html:1432)
>     at HTMLElement.__setProperty (polymer.html:1441)
>     at HTMLElement._applyEffectValue (polymer.html:1924)
>     at HTMLElement._annotationEffect (polymer.html:1587)
>     at HTMLElement._effectEffects (polymer.html:1448)
>     at HTMLElement._propertySetter (polymer.html:1432)

The error is in px-vis-behavior-common.html generateRandomID().
The line is: `this._uniqueIdsUsed.push(id);`

Theory:
I believe this is happening because the `PxVisBehavior.uniqueIds.detached()` method is being called which sets `_uniqueIdsUsed` to null.  When the px-vis-clip-path is created again it inherits the old data bindings instead of creating the new property.

I tried creating a codepen to reproduce the issue but polygit was not working.

This is the sequence of events where we see it:

- Create custom polymer component with <px-vis-timeseries> as an element.
  - px-vis-timeseries should NOT have height or width properties set.
- Create custom component using document.createElement()/.appendChild()

The timeseries properties look like this:
```
        <px-vis-timeseries 
          hidden={{middleError}}  
          display-threshold-title="true"
          include-all-series="true" 
          show-tooltip="true" 
          selection-type="xy" 
          chart-vertical-alignment="top"
          id="apmChart" 
          disable-navigator="{{!data.navigator}}"
          tooltip-config="{}" 
          threshold-data="{{threshold}}" 
          series-config="[[seriesConf]]" 
          register-config='{"type":"horizontal"}'
          chart-data="[[chartData]]"
          >
        </px-vis-timeseries>
```

The default values for the above properties look like this.
```
      data: {
        type: Object,
        notify: true,
        reflectToAttribute: true,
        observer: '_widgetDataChanged'
      },
      seriesConfig: {
        type: Object,
        notify: true
      },
      chartData: {
        type: Object,
        notify: true
      },
      threshold: {
        type: Array,
        notify: true
      },
      middleError: {
        type: Boolean,
        notify: true,
        value: false,
      }
```

Observations:
PxVisBehavior.uniqueIds.detached() (px-vis-behavior-common) is setting _uniqueIdsUsed to null (as expected).
The size observer in px-vis-clip-path.html is triggering when _uniqueIdsUsed is null.

This can be mitigated by specifying height and width properties e.g. `<px-vis-timeseries height="400" width="500" ...>`

Looking at the logic I'm still not entirely sure why this happens but it did block 2 teams for too long so it would be good to save time for anyone else who might encounter it.

Will be submitting a PR for possible fix.

**Comment by benoitjchevalier** [comment]
Hello,

thanks for taking the time of providing a detailed issue. I'm not entirely sure what the underlying problem is in your case but can't seem to be able to reproduce when moving charts around the dom. 

However taking the approach of resetting the array to an empty array instead of null when detaching sounds safe to me, so if that fixes your issue I'm going to release a patch version with that fix

**Comment by 212361828** [comment]
Thanks, much appreciated!

**Comment by benoitjchevalier** [comment]
Now available in vis v2.0.7
